src/Validator/Constraints/GiftCardIsNotExpiredValidator.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusGiftCardPlugin\Validator\Constraints;
+
+use Setono\SyliusGiftCardPlugin\Model\GiftCardInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+final class GiftCardIsNotExpiredValidator extends ConstraintValidator
+{
+    public function validate($value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof GiftCardIsNotExpired) {
+            throw new UnexpectedTypeException($constraint, GiftCardIsNotExpired::class);
+        }
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        /** @var GiftCardInterface $giftCard */
```

Comment:
Would be better to `Assert` this I guess?

Reply:
I also wanted the variable to be named $giftCard. So both would work, but we can assert as well :)